Batteries/Data/List/Lemmas.lean

```diff
@@ -92,17 +96,15 @@ theorem erase_eq_self_iff_forall_bne [BEq α] (a : α) (xs : List α) :
 
 /-! ### findIdx? -/
 
-theorem findIdx_eq_findIdx? (p : α → Bool) (l : List α) :
-    l.findIdx p = (match l.findIdx? p with | some i => i | none => l.length) := by grind
+@[deprecated (since := "2025-11-06")]
+alias findIdx_eq_findIdx? := findIdx_eq_getD_findIdx?
 
 /-! ### replaceF -/
 
 theorem replaceF_nil : [].replaceF p = [] := rfl
 
 theorem replaceF_cons (a : α) (l : List α) :
-    (a :: l).replaceF p = match p a with
-      | none => a :: replaceF p l
-      | some a' => a' :: l := rfl
+    (a :: l).replaceF p = (p a).elim (a :: replaceF p l) (· :: l) := by grind [replaceF]
```

Comment:
Is `elim` preferred over `match`? My inclination would be to leave this alone in the absence of a zulip discussion.